### PR TITLE
Add default script language completion

### LINKED ISSUE #1262: Allow configuration of default script language
**Is your feature request related to a problem? Please describe.**
Right now, when auto-importing a component, it just defaults to adding a "vanilla" JS script tag. I do all of my work in Svelte in TypeScript so I need to go back and add a `lang="ts"` to the script tag.

**Describe the solution you'd like**
It'd be awesome if there was a way to configure the default script language in the plugin's configuration. This would simply just add `lang="<lang>"` to the generated script tag when a component is imported. This could in theory be extended to other cases where script and even style tags are generated.

**Describe alternatives you've considered**
Could always just do it the way I'm doing now but this would definitely be a small productivity boost.

**Additional context**
I'd be willing to help out with this one if this is a change that you like 😄

### PR DESCRIPTION
**Pull request by devmattrick**
This PR fixes #1262 by adding a new configuration option: `svelte.plugin.svelte.defaultScriptLanguage`. This option can either be `none` or `ts`, which will configure Svelte VSCode to use `<script>` or `<script lang="ts">` respectively when generating new script tags on auto import.

Let me know if you'd like any changes. Thanks!

### CODE REVIEW

**Review by jasonlyu123 — COMMENTED** [review]
Looks good to me :+1:

Can you also document the config to the [README](https://github.com/sveltejs/language-tools/tree/master/packages/language-server/README.md)?

File: packages/language-server/src/plugins/typescript/features/CompletionProvider.ts
Comment (devmattrick) [reply packages/language-server/src/plugins/typescript/features/CompletionProvider.ts]: Done, not sure if I should have allowed undefined in the type in ls-config.

Comment (dummdidumm) [reply packages/language-server/src/plugins/typescript/features/CompletionProvider.ts]: I don't think this is necessary, could you undo this change? Inside `update` of the config manager the case of undefined values is already handled.

Comment (jasonlyu123) [reply packages/language-server/src/plugins/typescript/features/CompletionProvider.ts]: Yeah. I tested it it wasn't necessary. Sorry. It would only be falsy if it's explicitly set to be.

File: packages/svelte-vscode/package.json
Comment (dummdidumm) [inline packages/svelte-vscode/package.json:386]: Should we put this inside `completions`? Like `"svelte.plugin.svelte.completions.defaultScriptLanguage"`? @jasonlyu123 what do you think?

Comment (jasonlyu123) [reply packages/svelte-vscode/package.json:386]: Hmm. It is also being used in the code action. if we want to further organize it. Maybe the name needs to be more generic. Maybe something like `suggest` or `preferences`?

Comment (dummdidumm) [reply packages/svelte-vscode/package.json:386]: Oh right, it's also part of the code action. In that case, never mind 😄 My fear was that people would mistake this as something like "if I set this I don't need to write `<script lang="ts">` at all" (like the deprecated and discouraged default language of svelte preprocess).

**Comment by jrylan** [comment]
Thanks @devmattrick for getting this together. I just wanted to ask before a maintainer here merges: is this something that should instead go in `svelte.config.js`?

I see this issue https://github.com/sveltejs/language-tools/issues/422, would it be more straightforward to use that method instead?

```
// svelte.config.js
export default {
  preprocess: [
    autoPreprocess({
      defaults: { script: 'typescript' },
    }),
  ],
};
```

Does the VS Code extension pickup your svelte.config.js file?

**Comment by dummdidumm** [comment]
The config you mention is deprecated and discouraged. Problem is that not all tooling can reliably read the config, and there's also syntax highlighting limitations - as a result you get all kinds if weird behavior. This PR should only be about adding `lang="ts"` during a auto import or code action which created a script tag.

**Comment by jrylan** [comment]
@dummdidumm got it, thank you!

**Review by dummdidumm — APPROVED** [review]

**Comment by qwabra** [comment]
`.vscode/settings.json`
```
{
	"svelte.plugin.svelte.defaultScriptLanguage": "ts"
}
```
does not work?!

expexted `<script lang="ts"></script>`  
instead `<script></script>`
it is right?

**Comment by jasonlyu123** [comment]
nope. This PR is for the script tag created when there's an auto-import.

```svelte
<C
```

to

```svelte
<script lang="ts">
  import Component from 'path/to/Component.svelte'
</script>

<Component
```

**Comment by qwabra** [comment]
"an auto-import" hmm.

what about snippets, is that possible by Svelte config?

**Pull request merged**
